x/mongo/driver/topology/topology.go

```diff
@@ -478,7 +474,7 @@ func (t *Topology) selectServerFromDescription(desc description.Topology,
 
 	suitable, err := selectionState.selector.SelectServer(desc, allowed)
 	if err != nil {
-		return nil, wrapServerSelectionError(err, t)
+		return nil, ServerSelectionError{Wrapped: err, Desc: t.Description()}
```

Comment:
The `SelectServer` call used `desc`, so the error should use `desc` rather than `t.Description()`. Remember that there's a background routine that updates server and topology descriptions, so `t.Description()` might return something different than `desc`, which would be confusing for users.

Reply:
done